### ISSUE TITLE
Copy foreign-namespaced attributes in post-processors

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#447]
[Originally Ticket 1771]

In applications such as sTeX, we tend to introduce our own attributes, e.g. in a separate stex: namespace. To this extent, we extend the LaTeXML schema to allow those attributes.

However, in some cases we also need the post-processors to silently copy over foreign attributes. The current LaTeXML default behaviour is to strip away any attributes it hasn't been explicitly told to copy over (in the post-processors).

I have just implemented a temporary solution that respects foreign attributes of ltx:Math in the two math post-processors. To exemplify, here is one of the snippets in Post::OpenMath:

```
  # Using the prefix to determine foreign attributes
  my @foreign_attributes = grep { my $prefix = $_->prefix; $prefix && ($prefix !~ /^xml|ltx$/); } $math->attributes;
  my %foreign_copies = ();
  %foreign_copies = (map {$_->nodeName() => $_->getValue()} @foreign_attributes) if @foreign_attributes;

  my $wrapped = ['om:OMOBJ', \%foreign_copies, @conversion];
```

This can definitely be done better in general. Using the prefix is certainly worse than using the namespace URL, which is mandatory (but as I said, the above is a temporary hack, since I don't want to invest too much time at the moment. That is exactly why I am bookkeeping in this ticket).

Also, this treatment is conceivably something that would need to be applied on **all** elements, so there should be a general solution that isn't targeting specific elements. For sTeX we can make due with only the top-level Math element for now, but we can conceivably want to attach such attributes to XMToks and XMApps as well.

I am tentatively putting the ticket in the 0.9 milestone, so please ignore it for the moment.

**Comment by dginev** [comment]
Bruce is telling me that I should record a related problem to this issue:

Currently RDFa metadata recorded on any ltx:XMath node gets completely lost during post-processing to MathML. I would not consider those attributes "foreign", but it is conceivable we may need bits of the same infrastructure for both cases.

**Comment by kohlhase** [comment]
I very much support this copying to the end result. But while you are at it, we should not forget the OpenMath.

**Comment by dginev** [comment]
Actually, since you mention OpenMath, would copying RDFa attributes over to OpenMath be as simple as it is with MathML? RDFa is designed to be embeddable in HTML, so it "just works" when added to a MathML formula in an HTML document.

(I am currently looking at RDFa preservation in math during post-processing, which is familiar, but different, to the foreign namespace attributes)

**Comment by kohlhase** [comment]
I have not checked, but I am quite confident that RDFa attributes can be added to OM.

**Comment by dginev** [comment]
solved via #1960
